src/python/pants/backend/python/rules/importable_python_sources.py

```diff
@@ -33,19 +33,11 @@ class ImportablePythonSources:
 
 @rule
 async def prepare_python_sources(targets: Targets) -> ImportablePythonSources:
-    def is_relevant(tgt: Target) -> bool:
-        # NB: PythonRequirementsFileSources is a subclass of FilesSources. We filter it out so that
-        # requirements.txt is not included. If the user intended for the file to be included, they
-        # should use a normal `files()` target rather than `python_requirements()`.
-        return (
-            tgt.has_field(PythonSources)
-            or tgt.has_field(ResourcesSources)
-            or (tgt.has_field(FilesSources) and not tgt.has_field(PythonRequirementsFileSources))
-        )
-
     stripped_sources = await Get[SourceFiles](
         AllSourceFilesRequest(
-            (tgt.get(Sources) for tgt in targets if is_relevant(tgt)), strip_source_roots=True
+            (tgt.get(Sources) for tgt in targets),
+            valid_sources_types=(PythonSources, ResourcesSources, FilesSources),
+            strip_source_roots=True,
         )
```

Comment:
Once we add codegen, with this PR:

```python
         AllSourceFilesRequest(
             (tgt.get(Sources) for tgt in targets),
             valid_sources_types=(PythonSources, ResourcesSources, FilesSources),
             codegen_enabled=True,
             strip_source_roots=True,
         )
```

With the approach in https://github.com/pantsbuild/pants/pull/9634:

```python
        AllSourceFilesRequest(
            (
                tgt.get(Sources)
                for tgt in targets
                if any(
                    tgt.has_field(sources)
                    for sources in (PythonSources, ResourcesSources, FilesSources)
                )
                or isinstance(tgt.get(Sources), CodegenSources)
            ),
            codegen_language=PythonSources,
            strip_source_roots=True,
        )
```